tests/test_server_ops.py

```diff
@@ -130,3 +125,82 @@ async def test_server_ops_bootstrap_script(self):
                 self.fail(
                     'Unexpected server error output:\n' + stderr.decode()
                 )
+
+    async def test_server_ops_set_pg_max_connections(self):
+        bootstrap_command = (
+            r'CONFIGURE SYSTEM INSERT Auth '
+            r'{ priority := 0, method := (INSERT Trust) }'
+        )
+        actual = random.randint(50, 100)
+        async with tb.start_edgedb_server(
+            auto_shutdown=True,
+            bootstrap_command=bootstrap_command,
+            max_allowed_connections=actual,
+        ) as sd:
+            run_dir = sd.server_data['runstate_dir']
+            port = 0
+            for sock in pathlib.Path(run_dir).glob('.s.PGSQL.*'):
+                if sock.is_socket():
+                    port = int(sock.suffix[1:])
+                    break
+
+            con = await edgedb.async_connect(
+                user='edgedb', host=sd.host, port=sd.port)
+            self.assertEqual(await con.query_one('SELECT 1'), 1)
+
+            try:
+
+                conn = await asyncpg.connect(
+                    host=run_dir, port=port, user='postgres')
+                try:
+                    max_connectiosn = await conn.fetchval(
+                        "SELECT setting FROM pg_settings "
+                        "WHERE name = 'max_connections'")
+                    self.assertEqual(int(max_connectiosn), actual)
+                finally:
+                    await conn.close()
+
+            finally:
+                await con.aclose()
+
+    def test_server_ops_detect_postgres_pool_size(self):
+        port = edgedb_cluster.find_available_port()
+        actual = random.randint(50, 100)
+
+        async def test(host):
+            bootstrap_command = (
+                r'CONFIGURE SYSTEM INSERT Auth '
+                r'{ priority := 0, method := (INSERT Trust) }'
+            )
+            async with tb.start_edgedb_server(
+                auto_shutdown=True,
+                bootstrap_command=bootstrap_command,
+                max_allowed_connections=None,
+                postgres_dsn=f'postgres:///?user=postgres&port={port}&'
+                             f'host={host}',
+            ) as sd:
+                con = await edgedb.async_connect(
+                    user='edgedb', host=sd.host, port=sd.port)
+                try:
+                    max_connections = await con.query_one(
+                        'SELECT cfg::SystemConfig.__pg_max_connections '
+                        'LIMIT 1')
```

Comment:
Hm, does it work without `LIMIT 1`? If it doesn't it's a bug

Reply:
It's not a bug.  We don't have a notion of a singleton object, so the compiler is unable to prove that there is only one `cfg::SystemConfig`.

Reply:
Also, this reminded me that we need to implement cardinality assertions instead of using `LIMIT 1`